### ISSUE TITLE
Question on reproduction of NAFnet on UIEB

### ISSUE THREAD

**Issue by zzzack66** [zhoujingchun03/HCLR-Net#2]
Thanks for your awesome work.
Can you please provide the .yml file of NAFnet for the UIEB dataset? 
I tried several settings but couldn't get a PSNR value close to the one you provided.

**Comment by Hikari0608** [comment]
You can refer to #1  if you're asking about dataset partitioning

**Comment by zzzack66** [comment]
Thanks for your reply.
I'm asking for the reproduction of NAF-net on the UIEB dataset as shown in your paper. As shown in Table 1 NAF-net can get a result of 22.691 dB on the UIEB T-90 testing set. Can you please help me with the training configuration? Which should be a YAML file.
I tried several settings and couldn't get the result close to 22.6 dB.
I'm looking forward to your reply.

**Comment by Hikari0608** [comment]
I will convey your request to the collaborator, but he might be busy recently.

Perhaps you could share your configuration and the PSNR results you obtained, and I will try to assist you in resolving this issue.😊😊

**Comment by zzzack66** [comment]
name: NAFNet-UIEB-width64
model_type: ImageRestorationModel
scale: 1
num_gpu: 1
manual_seed: 10

datasets:
  train:
    name: gopro-train
    type: PairedImageDataset
    dataroot_gt: /home/UIEB/UIEB800/sharp_crops.lmdb
    dataroot_lq: /home/UIEB/UIEB800/blur_crops.lmdb

    filename_tmpl: '{}'
    io_backend:
      type: lmdb

    gt_size: 256
    use_flip: true
    use_rot: true

    # data loader
    use_shuffle: true
    num_worker_per_gpu: 8
    batch_size_per_gpu: 8
    dataset_enlarge_ratio: 1
    prefetch_mode: ~

  val:
    name: gopro-test
    type: PairedImageDataset
    dataroot_gt: /home/UIEB/UIEB90/sharp_crops.lmdb
    dataroot_lq: /home/UIEB/UIEB90/blur_crops.lmdb
    io_backend:
      type: lmdb


network_g:
  type: NAFNetLocal
  width: 64
  enc_blk_nums: [1, 1, 1, 28]
  middle_blk_num: 1
  dec_blk_nums: [1, 1, 1, 1]

# path
path:
  pretrain_network_g: ~
  strict_load_g: true
  resume_state: ~

# training settings
train:
  optim_g:
    type: AdamW
    lr: !!float 1e-3
    weight_decay: !!float 1e-3
    betas: [0.9, 0.9]

  scheduler:
    type: TrueCosineAnnealingLR
    T_max: 400000
    eta_min: !!float 1e-7

  total_iter: 400000
  warmup_iter: -1 # no warm up

  # losses
  pixel_opt:
    type: PSNRLoss
    loss_weight: 1
    reduction: mean

# validation settings
val:
  val_freq: !!float 2e4
  save_img: false


  metrics:
    psnr: # metric name, can be arbitrary
      type: calculate_psnr
      crop_border: 0
      test_y_channel: false
    ssim:
      type: calculate_ssim
      crop_border: 0
      test_y_channel: false

# logging settings
logger:
  print_freq: 200
  save_checkpoint_freq: !!float 5e3
  use_tb_logger: true
  wandb:
    project: ~
    resume_id: ~

# dist training settings
dist_params:
  backend: nccl
  port: 29500
  
  Here is the config I used. I got around 20dB on UIEB T-90. 
  Thanks for your time.

**Comment by Hikari0608** [comment]
In order to achieve the best performance of NAFNet, we conducted some additional porting work within the underwater image enhancement code framework, rather than performing the experiments in the official repository.

You can refer to the following settings to improve its accuracy🤗:

1. Ensure the UIEB dataset is trained for more than 150 epochs with a learning rate of 1e-4.
2. In the last 30 epochs, crop to 512*512 for the first 20 epochs, and revert to crop to 256*256 for the final 10 epochs.
